merkletree.go

```diff
@@ -121,11 +121,11 @@ func computePrivateIndex(key string) []byte {
 	return crypto.Digest([]byte(key))
```

Comment:
I think I agree with @arlolra, since if we change the `computePrivateIndex` then some tests would be broken. May @Liamsi  make a separate pull for that?